adapter/src/lib.rs

```diff
@@ -64,28 +65,55 @@ pub struct EthereumChannel {
     pub creator: String,
     pub token_addr: String,
     pub token_amount: String,
-    pub valid_until: String,
+    pub valid_until: i64,
     pub validators: String,
     pub spec: String,
 }
 
+impl From<&Channel> for EthereumChannel {
```

Comment:
this should be `TryFrom` and should handle errors like serialization failures gracefully

think about it - if it doesn't, then it's a DoS attack vector because anyone can pass an invalid spec and make the node crash